File: apps/medousa-home/src/lib/types/mcpGateway.ts
export interface McpServerConfig {
  id: string;
  title: string;
  enabled: boolean;
  transport: string;
  command?: string | null;
  args: string[];
  url?: string | null;
  bearerToken?: string | null;
  allowedLanes: string[];
  allowedEffectClasses: string[];
  useMock: boolean;
}

export interface McpGatewayFileConfig {
  gateway: {
    bind: string;
    daemonPolicyUrl: string;
    maxInvokeDurationMs: number;
    catalogRefreshIntervalSecs: number;
    useMockFallback: boolean;
  };
  servers: McpServerConfig[];
}

export interface McpGatewayConfigLoadResult {
  path: string;
  config: McpGatewayFileConfig;
  fileExists: boolean;
}

export interface McpGatewayHealth {
  status: string;
  invokesEnabled: boolean;
  registeredServers: number;
  connectedServers: number;
  catalogEntries: number;
}

export interface McpServerRuntime {
  serverId: string;
  title: string;
  enabled: boolean;
  connected: boolean;
  toolCount: number;
  allowedLanes: string[];
}

export interface McpGatewayStatusResult {
  gatewayUrl: string;
  reachable: boolean;
  message: string;
  health: McpGatewayHealth | null;
  servers: McpServerRuntime[];
  configPath: string;
}

export interface McpGatewayRestartResult {
  started: boolean;
  alreadyRunning: boolean;
  logPath: string;
  message: string;
}

export interface McpServerUpsertRequest {
  id: string;
  title: string;
  enabled?: boolean;
  transport?: string;
  command?: string | null;
  args?: string[];
  url?: string | null;
  bearerToken?: string | null;
  useMock?: boolean;
}

export interface McpServerMutationResult {
  ok: boolean;
  message: string;
  configPath: string;
}

export interface McpGatewayTestResult {
  ok: boolean;
  message: string;
  connected: boolean;
  toolCount: number;
}

File: python/medousa-sdk/src/medousa/types/mcp_gateway.py
from __future__ import annotations

from medousa.types.daemon_api import MedousaModel


class McpGatewayHealthSnapshot(MedousaModel):
    status: str
    invokes_enabled: bool
    registered_servers: int
    connected_servers: int
    catalog_entries: int


class McpGatewayServerRuntime(MedousaModel):
    server_id: str
    title: str
    enabled: bool
    connected: bool
    tool_count: int
    allowed_lanes: list[str]


class McpGatewayStatusResponse(MedousaModel):
    gateway_url: str
    reachable: bool
    message: str
    health: McpGatewayHealthSnapshot | None = None
    servers: list[McpGatewayServerRuntime] = []
